src/components/Main/Main.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react';
+import { Button, Input, Checkbox, Radiogroup } from '../common';
+import { TaskType } from "../types/types";
+import { v4 as uuidv4 } from 'uuid';
+import css from './main.module.css';
+
+export const Main = () => {
+
+    const todoList: TaskType[] = []
+    const filters = [
+        { id: '1', label: 'All', value: 'all' },
+        { id: '2', label: 'Active', value: 'active' },
+        { id: '3', label: 'Completed', value: 'completed' },
+    ]
+
+    const [newTask, setNewTask] = useState('')
+    const [tasks, setTasks] = useState(todoList)
+    const [filter, setFilter] = useState('all')
+    const [editedId, setEditedId] = useState<string | null>(null)
+    const [editTask, setEditTask] = useState('')
+
+    const addTaskHandler = () => {
+        if (newTask.trim() === '') {
+            return false
+        }
+        setTasks((prevTask) => [...prevTask, { id: uuidv4(), name: newTask, isDone: false }])
+        setNewTask('')
+    }
+    const deleteTaskHandler = (id: string) => {
+        setTasks(tasks.filter((task) => task.id !== id))
```

Comment:
стейт рассчитывается на основе предыдущего - используем синтаксис колбека
```suggestion
        setTasks((prevTask) => prevTask.filter((task) => task.id !== id))
```